src/Utilities/CssDocument.php

```diff
@@ -82,23 +61,47 @@ public function __construct(string $css)
      */
     public function getStyleRulesData(array $allowedMediaTypes): array
     {
-        $splitCss = $this->splitCssAndMediaQuery($allowedMediaTypes);
-
         $ruleMatches = [];
-        foreach ($splitCss as $cssPart) {
-            // process each part for selectors and definitions
-            \preg_match_all('/(?:^|[\\s^{}]*)([^{]+){([^}]*)}/mi', $cssPart['css'], $matches, PREG_SET_ORDER);
-
-            foreach ($matches as $cssRule) {
+        /** @var CssRenderable $rule */
+        foreach ($this->sabberwormCssDocument->getContents() as $rule) {
+            if ($rule instanceof CssAtRuleBlockList && $rule->atRuleName() === 'media') {
```

Comment:
Would a decicated method in sabberworm/php-css-parser make sense for this (i.e., should we open a ticket)?

Reply:
You mean for `$rule instanceof CssAtRuleBlockList`?  I think it's more than arguable that it doesn't justify a specific testing method, and that the code with the `instanceof` test is actually as succinct as it can be.

Also, for static analysis (e.g. Psalm) the `instanceof` test would probably still be needed even after a separate method returned `true`.